spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/DefaultReleaseManager.java

```diff
@@ -282,28 +286,22 @@ public Release status(Release release) {
 			}
 			release.getInfo().getStatus().setPlatformStatusAsAppStatusList(appStatusList);
 		}
-	return release;
+		return release;
 	}
 
 	public Release delete(Release release) {
 		AppDeployer appDeployer = this.deployerRepository.findByNameRequired(release.getPlatformName())
 				.getAppDeployer();
-
 		AppDeployerData appDeployerData = this.appDeployerDataRepository
 				.findByReleaseNameAndReleaseVersionRequired(release.getName(), release.getVersion());
 		List<String> deploymentIds = appDeployerData.getDeploymentIds();
 		if (!deploymentIds.isEmpty()) {
 			for (String deploymentId : deploymentIds) {
-
-				// don't error trying trying to undeploy something
-				// which is not deployed
-				AppStatus appStatus = appDeployer.status(deploymentId);
-				if (appStatus.getState().equals(DeploymentState.deployed)) {
+				try  {
 					appDeployer.undeploy(deploymentId);
 				}
-				else {
-					logger.warn("For Release name {}, did not undeploy existing app {} as its status is not "
-							+ "'deployed'.", release.getName(), deploymentId);
+				catch (Exception e) {
+					throw new SkipperException(e.getMessage());
```

Comment:
we should not be throwing an exception here, we need to iterate over all the IDs and log errors, otherwise we may end up in an inconsistent state with some apps not undeployed.

Reply:
oh yeah, I will update the PR.